File: JugglingLetters.cpp
#include <bits/stdc++.h>
using namespace std;

int main(void) {
    int t;
    cin >> t;
    while (t--) {
        vector<int> a(26);
        int n;
        cin >> n;
        for (int i=0; i<n; i++) {
            string s;
            cin >> s;
            for (char ch:s) 
                a[ch-'a']++;
        }
        bool ans = true;
        for (int i=0; i<n; i++) {
            if (a[i]%n!=0) {
                ans = false;
                break;
            }
        }
        if (ans)
            cout << "YES\n";
        else 
            cout << "NO\n";
    }

    return 0;
}

File: Python/JugglingLetters.py
def main():
    t = int(input())
    while t:
        t -= 1
        n = int(input())
        for i in range(n):
            s = input()
            a = [0 for i in range(26)]
            for j in s:
                a[ord(j)-ord('a')] += 1
        ans = True
        for i in range(n):
            if a[i] % n != 0:
                ans = False
                break
        if ans:
            print("YES")
        else:
            print("NO")


if __name__ == '__main__':
    main()
